### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by leon-vv**
Create a directory `generate_docs/` which contains code to generate documentation. To generate the documentation, use `bash ./make_docs.sh`. The version number has been bumped to `v0.9.0rc1` and documentation for this version has been commited. Open `docs/docs/traceon/index.html` in a browser to see the current state.

### CODE REVIEW

**Comment by jobdewitte** [comment]
Nice work, I like the overall feel of the documentation and usage also seems quite intuitive. I do have a few remarks and suggestions.

- Is there a way to make the classes and functions appear as a dropdown from the modules instead of on the bottom of the page? This would feel more natural than having to scroll down everytime, especially as the project grows.

- A few of the longer  class method names, especially in the `Exitation` module (e.g. `electrostatic_potential_at_point` are too long for the side bar on my 15.6'' laptop and have a single 't' on the next line. This feels a bit unprofessional? Is there a way to dynamically adjust this side bar? Otherwise I would suggest making it a bit wider.

- I would prefer a slightly more compact version of the documentation, which makes it a bit faster to navigate between functions. Maybe this is as easy as adapting the CSS to scale down the fontsize and whitespaces between functions.

- It would be nice to have code blocks that can be copied.
- More content related: some sort of general overview of the documentation, a how-to-start guide, explanation of the differences between Traceon and Traceon Pro  etc. should also be included in the docs, but I feel that is beyond the scope of this PR.

**Comment by leon-vv** [comment]
I would like to merge this so that we can move to the next release. I propose a new issue to collect your points which can not be tackled quickly: #180. Regarding your points:
(1) Quite difficult, would imply remaking the whole template (time intensive)
(2) Fixed
(3) Added to issue
(4) Added to issue
(5) Added to issue

**Review by jobdewitte — APPROVED** [review]

File: generate_docs/make_docs.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+VERSION=$(python3 -c "from importlib.metadata import version; print(version('traceon'))")
```
[the remark this answers is not in the document]
Comment (leon-vv) [reply generate_docs/make_docs.sh]: Fixed

File: generate_docs/make_docs.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+DIR=../docs/docs/v$VERSION/
+
+if [ -d $DIR ]; then
+    echo "Directory $DIR exists."
+	exit
+fi
+
+python3 ./custom_pdoc.py traceon -o $DIR  --force --html --config latex_math=True
```
[the remark this answers is not in the document]
Comment (leon-vv) [reply generate_docs/make_docs.sh]: Fixed

**Pull request merged**